app/uk/gov/hmrc/nrs/retrieval/config/HttpHead.scala

```diff
@@ -42,46 +43,47 @@ object CoreHttpReads extends HttpErrorFunctions {
 
   def responseHandler(method: String, url: String, response: HttpResponse): HttpResponse = {
     response.status match {
-      case status if (status == 404) => {
+      case status if status == NOT_FOUND =>
         logger.info(s"Submission bundle not found $status for query $method $url")
         if(method == HEAD_VERB) {
           response
         } else {
-          HttpResponse(404, JsArray.empty, Map[String,Seq[String]]())
+          HttpResponse(NOT_FOUND, JsArray.empty, Map[String,Seq[String]]())
         }
+      case _ => handleResponseEither(method, url)(response) match {
+        case Right(response) => response
+        case Left(err) => throw err
       }
-      case _ => handleResponse(method, url)(response)
     }
   }
 
   implicit val readRaw: CoreHttpReads[HttpResponse] = (method, url, response) => responseHandler(method, url, response)
 }
 
 trait CoreHead {
-  def HEAD[A](url: String)(implicit rds: CoreHttpReads[A], hc: HeaderCarrier, ec: ExecutionContext): Future[A]
+  def HEAD[A](url: String, headers: Seq[(String, String)])(implicit rds: CoreHttpReads[A], hc: HeaderCarrier, ec: ExecutionContext): Future[A]
```

Comment:
headers are now passed explicitly - another motivation for adding the it:tests prior to the upgrade.